### ISSUE TITLE
Add ssim_weight parameter to run_ad_recon and benchmark AD functions

### ISSUE THREAD

**Issue by cedriclim1** [cedriclim1/FastTomo#29]
## Background

A λ_ssim sweep on the benchmark phantom (200³, 29-angle, 1° tilt-axis error) shows that adding `tilt_ssim_loss` as an auxiliary AD loss term alongside TV consistently improves reconstruction quality:

| λ_ssim | SSIM vs phantom | NMSE |
|--------|----------------|------|
| 0 (baseline) | 0.9824 | 2.350e-02 |
| 1 | 0.9863 | 1.890e-02 |
| 10 | 0.9902 | 1.412e-02 |
| 100 | 0.9936 | 7.694e-03 |
| 1000 | 0.9941 | 7.017e-03 |

SSIM improves monotonically; NMSE drops 3.4×. Recommended operating point: **λ_ssim ≈ 100**. See `fasttomo-wiki/wiki/analyses/ssim-tilt-loss-benchmark-phantom.md` for full results and analysis.

Note: this effect is dataset-dependent. On the HEA atomic-resolution dataset, SSIM as a loss term *hurt* quality. It benefits dense, smoothly-varying objects (benchmark phantom) but not sparse atomic data.

## Work Required

`tilt_ssim_loss` is already implemented in `fasttomo/regularizers.py:105` and is differentiable. The change is to wire it in:

1. **`fasttomo/algorithms.py` — `run_ad_recon`**: add `ssim_weight: float = 0.0` parameter; add to the regularisation block:
   ```python
   if ssim_weight > 0.0:
       loss = loss + ssim_weight * tilt_ssim_loss(pred, sinogram)
   ```
   Update docstring.

2. **`benchmarks/ad_reconstruct.py`** — add `ssim_w: float = 0.0` to all four functions (`ad_reconstruct`, `ad_reconstruct_cached`, `ad_reconstruct_with_pose`, `ad_reconstruct_with_pose_cached`) and wire it into their loss lines.

3. **Tests**: add a test in `tests/test_cuda.py` that `run_ad_recon(..., ssim_weight=1.0)` runs without error and produces a finite loss.

4. **Wiki**: update `analyses/ssim-tilt-loss-benchmark-phantom.md` to replace the `#TODO` issue reference with the actual issue number.
